src/secureSignal_shared.ts

```diff
@@ -0,0 +1,122 @@
+const MAXIMUM_RETRY = 3;
+export class UidSecureSignalProvider {
+  debug: boolean;
+  isEuid: boolean;
+
+  constructor(debug = false, isEuid = false) {
+    this.debug = debug;
+    this.isEuid = isEuid;
+
+    if (
+      (!this.isEuid && typeof window.getUid2AdvertisingToken === 'function') ||
+      (this.isEuid && typeof window.getEuidAdvertisingToken === 'function')
+    ) {
+      this.logging('register SecureSignalProvider');
+      this.registerSecureSignalProvider();
+    }
+  }
+
+  public registerSecureSignalProvider = () => {
+    const uidHandler = this.retrieveAdvertisingTokenHandler();
+
+    if (!uidHandler) {
+      console.warn('UidSecureSignal: Please implement `getUidAdvertisingToken`');
+      return;
+    }
+
+    window.googletag = window.googletag || {
+      cmd: [],
+    };
+
+    window.googletag.secureSignalProviders = window.googletag.secureSignalProviders || [];
+    window.googletag.secureSignalProviders.push({
+      id: this.isEuid ? 'euid.eu' : 'uidapi.com',
+      collectorFunction: async () => {
+        this.logging('collectorFunction invoked');
+        const uidAdvertisingToken = await this.getUidAdvertisingTokenWithRetry(uidHandler);
+        this.logging(`collectorFunction pushes: ${uidAdvertisingToken}`);
+        return uidAdvertisingToken;
+      },
+    });
+  };
+
+  public logging = (message: string) => {
+    if (!this.debug) return;
+    console.log(`UidSecureSignal: ${message}`);
+  };
+
+  private retrieveAdvertisingTokenHandler = (): Function | undefined => {
+    if (this.isEuid) {
+      if (typeof window.getEuidAdvertisingToken === 'function') {
+        return window.getEuidAdvertisingToken!;
+      }
+      if (window.__euid && 'getAdvertisingTokenAsync' in window.__euid!) {
+        return window.__euid!.getAdvertisingTokenAsync!.bind(window.__euid);
+      }
+    } else {
+      if (typeof window.getUid2AdvertisingToken === 'function') {
+        return window.getUid2AdvertisingToken!;
+      }
+      if (window.__uid2 && 'getAdvertisingTokenAsync' in window.__uid2!) {
+        return window.__uid2!.getAdvertisingTokenAsync!.bind(window.__uid2);
+      }
+    }
+
+    // if (window.__uid2 && 'getAdvertisingTokenAsync' in window.__uid2!) {
```

Comment:
I'll throw in a quick commit to remove this commented-out code (and below as well).